src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -677,6 +677,11 @@ private void LookupForMatchingGitHubRepository(Manifest manifest)
                 }
                 else
                 {
+                    if (response.StatusCode == System.Net.HttpStatusCode.Forbidden
```

Comment:
The forbidden case is weird - if we mess up the token, we will think it's rate limiting? Can we maybe log the status code and say it failed that way?

Reply:
Maybe there's something in the response body to tell?

Reply:
I put the 403 there because documentation says it can return it too https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#:~:text=If%20you%20exceed%20your%20primary%20rate%20limit%2C%20you%20will%20receive%20a%20403%20or%20429%20response%2C%20and%20the%20x%2Dratelimit%2Dremaining%20header%20will%20be%200.%20You%20should%20not%20retry%20your%20request%20until%20after%20the%20time%20specified%20by%20the%20x%2Dratelimit%2Dreset%20header. I'll add the response body